src/components/IntegrationsView/IntegrationForm/IntegrationForm.tsx

```diff
@@ -277,31 +277,24 @@ const SmartLink: FC<{
     }
     if (!sl_chat) return;
 
-    const messages = sl_chat.reduce<ChatMessages>((acc, message) => {
-      if (message.role === "user" && typeof message.content === "string") {
-        return [
-          ...acc,
-          {
-            role: message.role,
-            content: message.content,
-          },
-        ];
-      }
+    const messages = formatMessagesForChat(sl_chat);
 
-      // TODO: Other types.
-      return acc;
-    }, []);
-    // dispatch(newChatAction()); id is out of date
-    dispatch(setToolUse("agent"));
```

Comment:
We might need configuration chat to run with agentic tool use, otherwise, it won't be able to run integration tools within a chat